specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,1002 @@
+        // even though the version string in the registry may be stored as x.x if the full version is
+        // x.x.0.0 
+        auto versionPartsFound = std::count(productVersionString.begin(), productVersionString.end(), '.') + 1;
+        INT64 versionPartsRequired = 4;
+        INT64 versionDigitsToAdd = versionPartsRequired - versionPartsFound;
+        for (auto i = 0; i < versionDigitsToAdd; i++)
+        {
+            productVersionString += L".0";
+        }
+        std::wstring processorArchitecture{};
+        SYSTEM_INFO systemInfo;
+        GetNativeSystemInfo(&systemInfo);
+        switch (systemInfo.wProcessorArchitecture)
+        {
+        case PROCESSOR_ARCHITECTURE_AMD64:
+            processorArchitecture += L"x64";
+            break;
+        case PROCESSOR_ARCHITECTURE_ARM:
+            processorArchitecture += L"arm";
+            break;
+        case PROCESSOR_ARCHITECTURE_ARM64:
+            processorArchitecture += L"arm64";
+            break;
+        case PROCESSOR_ARCHITECTURE_INTEL:
+            processorArchitecture += L"x86";
+            break;
+        case PROCESSOR_ARCHITECTURE_IA64:
+        case PROCESSOR_ARCHITECTURE_UNKNOWN:
+            winrt::throw_hresult(E_UNEXPECTED);
+        }
+
+        std::filesystem::path platformSDKExecutablePath = installPathString;
+        platformSDKExecutablePath /= L"bin";
+        platformSDKExecutablePath /= productVersionString;
+        platformSDKExecutablePath /= processorArchitecture;
+
+        return platformSDKExecutablePath;
+    }
+
+    void LaunchSignToolWithArguments(std::wstring arguments)
+    {
+        std::filesystem::path signToolPath = GetPlatformSDKPath();
+        signToolPath /= L"signtool.exe";
+
+        std::wstring commandLine{ signToolPath };
+        commandLine += L" " + arguments;
+        DWORD exitCode{};
+        winrt::check_hresult(CreateProcessAndWaitForExitCode(commandLine, exitCode));
+        if (exitCode == EXIT_FAILURE)
+        {
+            winrt::throw_hresult(E_FAIL);
+        }
+    }
+
+    void Example_SignPackage(std::wstring pfxFile, std::wstring packageFile)
+    {
+        std::wstring signToolArguments;
+        signToolArguments.append(L"sign /fd SHA256 /f ");
+        signToolArguments.append(pfxFile);
+        signToolArguments.append(L" " + packageFile);
+        LaunchSignToolWithArguments(signToolArguments);
+    }
+```
+
+# 3. API Details
+
+```c#
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+namespace Microsoft.Kozani.MakeMSIX
+{
+    [contractversion(1)]
+    apicontract MakeMSIXContract{};
+
+    /// Options for creating a package
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass CreatePackageOptions
+    {
+        /// Created by callers to pass options into CreatePackage()
+        CreatePackageOptions();
+
+        /// Optional replacement package name.
```

Comment:
SUGGESTION: `/// Optional override for package name`

same comment applies to other replacement/override properties